File: lib/models/medical_products_entity.dart
class MedicalProductsEntity {
  final String id;
  final String title;
  final String description;
  final String category;
  final dynamic price;      // keep dynamic; parse to double when showing
  final dynamic oldPrice;
  final String? discount;
  final String? img;
  final String? createdAt;

  MedicalProductsEntity({
    required this.id,
    required this.title,
    required this.description,
    required this.category,
    required this.price,
    this.oldPrice,
    this.discount,
    this.img,
    this.createdAt,
  });

  factory MedicalProductsEntity.fromFirestore(String id, Map<String, dynamic> data) {
    return MedicalProductsEntity(
      id: id,
      title: (data['title'] ?? '').toString(),
      description: (data['description'] ?? '').toString(),
      category: (data['category'] ?? '').toString(),
      price: data['price'],
      oldPrice: data['oldPrice'],
      discount: data['discount']?.toString(),
      img: data['img']?.toString(),
      createdAt: data['createdAt']?.toString(),
    );
  }

  static String generateProductId(MedicalProductsEntity p) => p.id;
}

File: src/entity/medicalproductsEntity.js
export class MedicalProductsEntity {
  constructor({ 
    id = null, 
    title, 
    description, 
    category, 
    price, 
    oldPrice = null, 
    discount = null, 
    img, 
    createdAt = new Date().toISOString() 
  }) {
    this.id = id;
    this.title = title;
    this.description = description;
    this.category = category;
    this.price = price;
    this.oldPrice = oldPrice;
    this.discount = discount;
    this.img = img;
    this.createdAt = createdAt;
  }

  // Helper method to generate product ID
  static generateProductId(product) {
    return product.title.toLowerCase().replace(/[^a-z0-9]/g, '-');
  }

  // Helper method to extract price value
  static extractPrice(priceText) {
    const priceMatch = priceText.match(/S\$(\d+\.?\d*)/);
    return priceMatch ? parseFloat(priceMatch[1]) : 0.00;
  }
}